android/src/main/java/ly/count/android/sdk/react/CountlyReactNative.java

```diff
@@ -1727,6 +1629,41 @@ static void log(String message, Throwable tr, LogLevel logLevel) {
         }
     }
 
+    public Map<String, Object> convertToEventMap(ReadableArray segments) {
+        Map<String, Object> segmentation = new HashMap<>();
+        if (segments == null) {
+            return segmentation;
+        }
+        int len = segments.size();
+        for (int i = 0; i < len; i += 2) {
+            String key = segments.getString(i);
+            if (i + 1 < len) {
+                ReadableType valueType = segments.getType(i + 1);
+                switch (valueType) {
+                    case String:
+                        segmentation.put(key, segments.getString(i + 1));
+                        break;
+                    case Boolean:
+                        segmentation.put(key, segments.getBoolean(i + 1));
+                        break;
+                    case Number:
+                        double doubleValue = segments.getDouble(i + 1);
+                        int intValue = (int) doubleValue; // casting to int will remove the decimal part
+                        if (doubleValue == intValue) {
```

Comment:
It is just a thought, is it needed to check for this. Because at the end, if it is convertible to the int Android side will convert it via JSONObject